stdsimd-verify/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+#![feature(proc_macro)]
+
+extern crate proc_macro;
+extern crate proc_macro2;
+extern crate syn;
+#[macro_use]
+extern crate quote;
+
+use std::path::Path;
+use std::fs::File;
+use std::io::Read;
+
+use proc_macro::TokenStream;
+use quote::Tokens;
+
+macro_rules! my_quote {
+    ($($t:tt)*) => (quote_spanned!(proc_macro2::Span::call_site(), $($t)*))
+}
+
+#[proc_macro]
+pub fn x86_functions(input: TokenStream) -> TokenStream {
+    let dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+    let root = dir.parent().unwrap();
+    let root = root.join("coresimd/src/x86");
+
+    let mut files = Vec::new();
+    walk(&root, &mut files);
+
+    let mut functions = Vec::new();
+    for file in files {
+        for item in file.items {
+            match item {
+                syn::Item::Fn(f) => functions.push(f),
+                _ => {}
+            }
+        }
+    }
+
+    functions.retain(|f| {
+        match f.vis {
+            syn::Visibility::Public(_) => {}
+            _ => return false,
+        }
+        match f.unsafety {
+            syn::Unsafety::Unsafe(_) => {}
+            _ => return false,
```

Comment:
Are there `pub` functions in the `x86` module that are not `unsafe`? What are those?

Reply:
Presumably not because we said that all intrinsics should be unsafe.

Reply:
Yes I think so too, maybe we should emit a warning here or something, but that can be done in a future PR.